cypress/e2e/permit2.test.ts

```diff
@@ -0,0 +1,212 @@
+import { MaxUint160, MaxUint256 } from '@uniswap/permit2-sdk'
+
+import { DAI, USDC_MAINNET } from '../../src/constants/tokens'
+
+const APPROVE_BUTTON = '[data-testid="swap-approve-button"]'
+
+// The same tokens & swap-amount combination is used for all permit2 tests
+const INPUT_TOKEN = DAI
+const OUTPUT_TOKEN = USDC_MAINNET
+const TEST_BALANCE_INCREMENT = 0.01
+
+const setupSwap = () => {
+  // Use input/output in search params for faster test setup
+  cy.visit(`/swap/?inputCurrency=${INPUT_TOKEN.address}&outputCurrency=${OUTPUT_TOKEN.address}`, {
```

Comment:
I disagree. Since a swap is setup before every permit2 test, the fastest setup method should be used (url params). `swap.test.ts` should ensure there is no difference between manual vs url param swap setup, so `permit2.test.ts`'s usage of one vs the other is arbitrary, other than speed